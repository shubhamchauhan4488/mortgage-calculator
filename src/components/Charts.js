import React from "react";
import {
  AreaChart,
  Area,
  CartesianGrid,
  XAxis,
  YAxis,
  Tooltip,
  Bar,
  BarChart,
  Legend,
} from "recharts";
import { colors } from "../globals/theme";

export const Charts = (props) => {
  return (
    <div className="container-fluid">
      <h2 className="diagram-heading">Payment Diagrams</h2>
      <div className="row justify-content-center justify-content-md-around mb-3 mt-3">
        <AreaChart
          className='chart-container'
          width={500}
          height={250}
          data={props.graphData}
        >
          <defs>
            <linearGradient id="colorUv" x1="0" y1="0" x2="0" y2="1">
              <stop offset="5%" stopColor={colors.PURPLE} stopOpacity={0.8} />
              <stop offset="95%" stopColor={colors.PURPLE} stopOpacity={0.1} />
            </linearGradient>
          </defs>
          <XAxis dataKey="numberOfPayments" tick={{ stroke: colors.PRIMARY }} />
          <YAxis tick={{ stroke: colors.PRIMARY }} />
          <CartesianGrid strokeDasharray="3 3" />
          <Tooltip />
          <Area
            type="monotone"
            dataKey="principal"
            stroke={colors.PURPLE}
            fillOpacity={1}
            fill="url(#colorUv)"
          />
          <Area
            type="monotone"
            dataKey="numberOfPayments"
            stroke={colors.LIGHT_GREEN}
            fillOpacity={0}
            fill="url(#colorPv)"
          />
        </AreaChart>

        <BarChart
          className='chart-container'
          width={300}
          height={250}
          data={props.barData}
        >
          <CartesianGrid strokeDasharray="3 3" />
          <XAxis dataKey="totalCost" tick={{ stroke: colors.PRIMARY }} />
          <YAxis tick={{ stroke: colors.PRIMARY }} />
          <Tooltip />
          <Legend wrapperStyle={{ color: "white" }} />
          <Bar dataKey="principal" fill={colors.PURPLE} />
          <Bar dataKey="interest" fill={colors.LIGHT_GREEN} />
        </BarChart>
      </div>
    </div>
  );
};
