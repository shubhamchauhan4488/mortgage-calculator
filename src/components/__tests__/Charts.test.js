import React from 'react';
import { cleanup, render } from '@testing-library/react';
import { Charts } from '../Charts';

describe('Summary Table', () => {
  afterEach(cleanup);
  const mockData = [
    {
      numberOfPayments: "0",
      principal: 50000
    },
    {
      numberOfPayments: "12",
      principal: 48872.5
    },
    {
      numberOfPayments: "24",
      principal: 47776.83
    },
    {
      numberOfPayments: "36",
      principal: 46625.68
    }
  ]

  it('renders correctly', () => {
    const { asFragment } = render(<Charts graphData={mockData} />);
    expect(asFragment()).toMatchSnapshot();
  });
})