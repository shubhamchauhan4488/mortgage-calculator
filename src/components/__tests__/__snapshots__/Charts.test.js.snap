// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary Table renders correctly 1`] = `
<DocumentFragment>
  <div
    class="container-fluid"
  >
    <h2
      class="diagram-heading"
    >
      Payment Diagrams
    </h2>
    <div
      class="row justify-content-center justify-content-md-around mb-3 mt-3"
    >
      <div
        class="recharts-wrapper chart-container"
        style="position: relative; cursor: default; width: 500px; height: 250px;"
      >
        <svg
          class="recharts-surface"
          height="250"
          version="1.1"
          viewBox="0 0 500 250"
          width="500"
        >
          <defs>
            <clippath
              id="recharts3-clip"
            >
              <rect
                height="210"
                width="430"
                x="65"
                y="5"
              />
            </clippath>
          </defs>
          <defs>
            <lineargradient
              id="colorUv"
              x1="0"
              x2="0"
              y1="0"
              y2="1"
            >
              <stop
                offset="5%"
                stop-color="#8884d8"
                stop-opacity="0.8"
              />
              <stop
                offset="95%"
                stop-color="#8884d8"
                stop-opacity="0.1"
              />
            </lineargradient>
          </defs>
          <g
            class="recharts-layer recharts-cartesian-axis recharts-xAxis xAxis"
          >
            <line
              class="recharts-cartesian-axis-line"
              fill="none"
              height="30"
              stroke="#666"
              width="430"
              x="65"
              x1="65"
              x2="495"
              y="215"
              y1="215"
              y2="215"
            />
            <g
              class="recharts-cartesian-axis-ticks"
            >
              <g
                class="recharts-layer recharts-cartesian-axis-tick"
              >
                <line
                  class="recharts-cartesian-axis-tick-line"
                  fill="none"
                  height="30"
                  stroke="#666"
                  width="430"
                  x="65"
                  x1="65"
                  x2="65"
                  y="215"
                  y1="221"
                  y2="215"
                />
                <text
                  class="recharts-text recharts-cartesian-axis-tick-value"
                  fill="#666"
                  height="30"
                  stroke="#e9ecef"
                  text-anchor="middle"
                  width="430"
                  x="65"
                  y="223"
                >
                  <tspan
                    dy="0.71em"
                    x="65"
                  >
                    0
                  </tspan>
                </text>
              </g>
              <g
                class="recharts-layer recharts-cartesian-axis-tick"
              >
                <line
                  class="recharts-cartesian-axis-tick-line"
                  fill="none"
                  height="30"
                  stroke="#666"
                  width="430"
                  x="65"
                  x1="208.33333333333334"
                  x2="208.33333333333334"
                  y="215"
                  y1="221"
                  y2="215"
                />
                <text
                  class="recharts-text recharts-cartesian-axis-tick-value"
                  fill="#666"
                  height="30"
                  stroke="#e9ecef"
                  text-anchor="middle"
                  width="430"
                  x="208.33333333333334"
                  y="223"
                >
                  <tspan
                    dy="0.71em"
                    x="208.33333333333334"
                  >
                    12
                  </tspan>
                </text>
              </g>
              <g
                class="recharts-layer recharts-cartesian-axis-tick"
              >
                <line
                  class="recharts-cartesian-axis-tick-line"
                  fill="none"
                  height="30"
                  stroke="#666"
                  width="430"
                  x="65"
                  x1="351.6666666666667"
                  x2="351.6666666666667"
                  y="215"
                  y1="221"
                  y2="215"
                />
                <text
                  class="recharts-text recharts-cartesian-axis-tick-value"
                  fill="#666"
                  height="30"
                  stroke="#e9ecef"
                  text-anchor="middle"
                  width="430"
                  x="351.6666666666667"
                  y="223"
                >
                  <tspan
                    dy="0.71em"
                    x="351.6666666666667"
                  >
                    24
                  </tspan>
                </text>
              </g>
              <g
                class="recharts-layer recharts-cartesian-axis-tick"
              >
                <line
                  class="recharts-cartesian-axis-tick-line"
                  fill="none"
                  height="30"
                  stroke="#666"
                  width="430"
                  x="65"
                  x1="495"
                  x2="495"
                  y="215"
                  y1="221"
                  y2="215"
                />
                <text
                  class="recharts-text recharts-cartesian-axis-tick-value"
                  fill="#666"
                  height="30"
                  stroke="#e9ecef"
                  text-anchor="middle"
                  width="430"
                  x="495"
                  y="223"
                >
                  <tspan
                    dy="0.71em"
                    x="495"
                  >
                    36
                  </tspan>
                </text>
              </g>
            </g>
          </g>
          <g
            class="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
          >
            <line
              class="recharts-cartesian-axis-line"
              fill="none"
              height="210"
              stroke="#666"
              width="60"
              x="5"
              x1="65"
              x2="65"
              y="5"
              y1="5"
              y2="215"
            />
            <g
              class="recharts-cartesian-axis-ticks"
            >
              <g
                class="recharts-layer recharts-cartesian-axis-tick"
              >
                <line
                  class="recharts-cartesian-axis-tick-line"
                  fill="none"
                  height="210"
                  stroke="#666"
                  width="60"
                  x="5"
                  x1="59"
                  x2="65"
                  y="5"
                  y1="215"
                  y2="215"
                />
                <text
                  class="recharts-text recharts-cartesian-axis-tick-value"
                  fill="#666"
                  height="210"
                  stroke="#e9ecef"
                  text-anchor="end"
                  width="60"
                  x="57"
                  y="215"
                >
                  <tspan
                    dy="0.355em"
                    x="57"
                  >
                    0
                  </tspan>
                </text>
              </g>
              <g
                class="recharts-layer recharts-cartesian-axis-tick"
              >
                <line
                  class="recharts-cartesian-axis-tick-line"
                  fill="none"
                  height="210"
                  stroke="#666"
                  width="60"
                  x="5"
                  x1="59"
                  x2="65"
                  y="5"
                  y1="162.5"
                  y2="162.5"
                />
                <text
                  class="recharts-text recharts-cartesian-axis-tick-value"
                  fill="#666"
                  height="210"
                  stroke="#e9ecef"
                  text-anchor="end"
                  width="60"
                  x="57"
                  y="162.5"
                >
                  <tspan
                    dy="0.355em"
                    x="57"
                  >
                    15000
                  </tspan>
                </text>
              </g>
              <g
                class="recharts-layer recharts-cartesian-axis-tick"
              >
                <line
                  class="recharts-cartesian-axis-tick-line"
                  fill="none"
                  height="210"
                  stroke="#666"
                  width="60"
                  x="5"
                  x1="59"
                  x2="65"
                  y="5"
                  y1="110"
                  y2="110"
                />
                <text
                  class="recharts-text recharts-cartesian-axis-tick-value"
                  fill="#666"
                  height="210"
                  stroke="#e9ecef"
                  text-anchor="end"
                  width="60"
                  x="57"
                  y="110"
                >
                  <tspan
                    dy="0.355em"
                    x="57"
                  >
                    30000
                  </tspan>
                </text>
              </g>
              <g
                class="recharts-layer recharts-cartesian-axis-tick"
              >
                <line
                  class="recharts-cartesian-axis-tick-line"
                  fill="none"
                  height="210"
                  stroke="#666"
                  width="60"
                  x="5"
                  x1="59"
                  x2="65"
                  y="5"
                  y1="57.5"
                  y2="57.5"
                />
                <text
                  class="recharts-text recharts-cartesian-axis-tick-value"
                  fill="#666"
                  height="210"
                  stroke="#e9ecef"
                  text-anchor="end"
                  width="60"
                  x="57"
                  y="57.5"
                >
                  <tspan
                    dy="0.355em"
                    x="57"
                  >
                    45000
                  </tspan>
                </text>
              </g>
              <g
                class="recharts-layer recharts-cartesian-axis-tick"
              >
                <line
                  class="recharts-cartesian-axis-tick-line"
                  fill="none"
                  height="210"
                  stroke="#666"
                  width="60"
                  x="5"
                  x1="59"
                  x2="65"
                  y="5"
                  y1="5"
                  y2="5"
                />
                <text
                  class="recharts-text recharts-cartesian-axis-tick-value"
                  fill="#666"
                  height="210"
                  stroke="#e9ecef"
                  text-anchor="end"
                  width="60"
                  x="57"
                  y="5"
                >
                  <tspan
                    dy="0.355em"
                    x="57"
                  >
                    60000
                  </tspan>
                </text>
              </g>
            </g>
          </g>
          <g
            class="recharts-cartesian-grid"
          >
            <g
              class="recharts-cartesian-grid-horizontal"
            >
              <line
                fill="none"
                height="210"
                stroke="#ccc"
                stroke-dasharray="3 3"
                width="430"
                x="65"
                x1="65"
                x2="495"
                y="5"
                y1="215"
                y2="215"
              />
              <line
                fill="none"
                height="210"
                stroke="#ccc"
                stroke-dasharray="3 3"
                width="430"
                x="65"
                x1="65"
                x2="495"
                y="5"
                y1="162.5"
                y2="162.5"
              />
              <line
                fill="none"
                height="210"
                stroke="#ccc"
                stroke-dasharray="3 3"
                width="430"
                x="65"
                x1="65"
                x2="495"
                y="5"
                y1="110"
                y2="110"
              />
              <line
                fill="none"
                height="210"
                stroke="#ccc"
                stroke-dasharray="3 3"
                width="430"
                x="65"
                x1="65"
                x2="495"
                y="5"
                y1="57.5"
                y2="57.5"
              />
              <line
                fill="none"
                height="210"
                stroke="#ccc"
                stroke-dasharray="3 3"
                width="430"
                x="65"
                x1="65"
                x2="495"
                y="5"
                y1="5"
                y2="5"
              />
            </g>
            <g
              class="recharts-cartesian-grid-vertical"
            >
              <line
                fill="none"
                height="210"
                stroke="#ccc"
                stroke-dasharray="3 3"
                width="430"
                x="65"
                x1="65"
                x2="65"
                y="5"
                y1="5"
                y2="215"
              />
              <line
                fill="none"
                height="210"
                stroke="#ccc"
                stroke-dasharray="3 3"
                width="430"
                x="65"
                x1="208.33333333333334"
                x2="208.33333333333334"
                y="5"
                y1="5"
                y2="215"
              />
              <line
                fill="none"
                height="210"
                stroke="#ccc"
                stroke-dasharray="3 3"
                width="430"
                x="65"
                x1="351.6666666666667"
                x2="351.6666666666667"
                y="5"
                y1="5"
                y2="215"
              />
              <line
                fill="none"
                height="210"
                stroke="#ccc"
                stroke-dasharray="3 3"
                width="430"
                x="65"
                x1="495"
                x2="495"
                y="5"
                y1="5"
                y2="215"
              />
            </g>
          </g>
          <g
            class="recharts-layer recharts-area"
          >
            <g
              class="recharts-layer"
            >
              <defs>
                <clippath
                  id="animationClipPath-recharts-area-4"
                >
                  <rect
                    height="216"
                    width="0"
                    x="65"
                    y="0"
                  />
                </clippath>
              </defs>
              <g
                class="recharts-layer"
                clip-path="url(#animationClipPath-recharts-area-4)"
              >
                <g
                  class="recharts-layer"
                >
                  <path
                    class="recharts-curve recharts-area-area"
                    d="M65,39.99999999999999C112.77777777777777,41.32470041666666,160.55555555555557,42.64940083333332,208.33333333333334,43.946249999999985C256.11111111111114,45.24309916666665,303.8888888888889,46.470449999999985,351.6666666666667,47.781094999999986C399.44444444444446,49.09173999999999,447.22222222222223,50.45092999999999,495,51.81012L495,215C447.22222222222223,215,399.44444444444446,215,351.6666666666667,215C303.8888888888889,215,256.11111111111114,215,208.33333333333334,215C160.55555555555557,215,112.77777777777777,215,65,215Z"
                    fill="url(#colorUv)"
                    fill-opacity="1"
                    height="210"
                    stroke="none"
                    width="430"
                  />
                  <path
                    class="recharts-curve recharts-area-curve"
                    d="M65,39.99999999999999C112.77777777777777,41.32470041666666,160.55555555555557,42.64940083333332,208.33333333333334,43.946249999999985C256.11111111111114,45.24309916666665,303.8888888888889,46.470449999999985,351.6666666666667,47.781094999999986C399.44444444444446,49.09173999999999,447.22222222222223,50.45092999999999,495,51.81012"
                    fill="none"
                    fill-opacity="1"
                    height="210"
                    stroke="#8884d8"
                    width="430"
                  />
                </g>
              </g>
            </g>
          </g>
          <g
            class="recharts-layer recharts-area"
          >
            <g
              class="recharts-layer"
            >
              <defs>
                <clippath
                  id="animationClipPath-recharts-area-5"
                >
                  <rect
                    height="216"
                    width="0"
                    x="65"
                    y="0"
                  />
                </clippath>
              </defs>
              <g
                class="recharts-layer"
                clip-path="url(#animationClipPath-recharts-area-5)"
              >
                <g
                  class="recharts-layer"
                >
                  <path
                    class="recharts-curve recharts-area-area"
                    d="M65,215C112.77777777777777,214.986,160.55555555555557,214.97199999999998,208.33333333333334,214.958C256.11111111111114,214.94400000000002,303.8888888888889,214.93000000000004,351.6666666666667,214.91600000000003C399.44444444444446,214.90200000000002,447.22222222222223,214.888,495,214.87399999999997L495,215C447.22222222222223,215,399.44444444444446,215,351.6666666666667,215C303.8888888888889,215,256.11111111111114,215,208.33333333333334,215C160.55555555555557,215,112.77777777777777,215,65,215Z"
                    fill="url(#colorPv)"
                    fill-opacity="0"
                    height="210"
                    stroke="none"
                    width="430"
                  />
                  <path
                    class="recharts-curve recharts-area-curve"
                    d="M65,215C112.77777777777777,214.986,160.55555555555557,214.97199999999998,208.33333333333334,214.958C256.11111111111114,214.94400000000002,303.8888888888889,214.93000000000004,351.6666666666667,214.91600000000003C399.44444444444446,214.90200000000002,447.22222222222223,214.888,495,214.87399999999997"
                    fill="none"
                    fill-opacity="0"
                    height="210"
                    stroke="#82ca9d"
                    width="430"
                  />
                </g>
              </g>
            </g>
          </g>
        </svg>
        <div
          class="recharts-tooltip-wrapper"
          style="transform: translate(undefinedpx, undefinedpx); pointer-events: none; visibility: hidden; position: absolute; top: 0px;"
        >
          <div
            class="recharts-default-tooltip"
            style="margin: 0px; padding: 10px; background-color: rgb(255, 255, 255); border: 1px solid #ccc; white-space: nowrap;"
          >
            <p
              class="recharts-tooltip-label"
              style="margin: 0px;"
            />
          </div>
        </div>
      </div>
      <div
        class="recharts-wrapper chart-container"
        style="position: relative; cursor: default; width: 300px; height: 250px;"
      >
        <svg
          class="recharts-surface"
          height="250"
          version="1.1"
          viewBox="0 0 300 250"
          width="300"
        >
          <defs>
            <clippath
              id="recharts6-clip"
            >
              <rect
                height="210"
                width="230"
                x="65"
                y="5"
              />
            </clippath>
          </defs>
          <g
            class="recharts-cartesian-grid"
          >
            <g
              class="recharts-cartesian-grid-horizontal"
            >
              <line
                fill="none"
                height="210"
                stroke="#ccc"
                stroke-dasharray="3 3"
                width="230"
                x="65"
                x1="65"
                x2="295"
                y="5"
                y1="5"
                y2="5"
              />
              <line
                fill="none"
                height="210"
                stroke="#ccc"
                stroke-dasharray="3 3"
                width="230"
                x="65"
                x1="65"
                x2="295"
                y="5"
                y1="215"
                y2="215"
              />
            </g>
            <g
              class="recharts-cartesian-grid-vertical"
            >
              <line
                fill="none"
                height="210"
                stroke="#ccc"
                stroke-dasharray="3 3"
                width="230"
                x="65"
                x1="65"
                x2="65"
                y="5"
                y1="5"
                y2="215"
              />
              <line
                fill="none"
                height="210"
                stroke="#ccc"
                stroke-dasharray="3 3"
                width="230"
                x="65"
                x1="295"
                x2="295"
                y="5"
                y1="5"
                y2="215"
              />
            </g>
          </g>
          <g
            class="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
          >
            <line
              class="recharts-cartesian-axis-line"
              fill="none"
              height="210"
              stroke="#666"
              width="60"
              x="5"
              x1="65"
              x2="65"
              y="5"
              y1="5"
              y2="215"
            />
            <g
              class="recharts-cartesian-axis-ticks"
            />
          </g>
        </svg>
        <div
          class="recharts-legend-wrapper"
          style="position: absolute; width: 290px; height: auto; left: 5px; bottom: 5px; color: white;"
        >
          <ul
            class="recharts-default-legend"
            style="padding: 0px; margin: 0px; text-align: center;"
          >
            <li
              class="recharts-legend-item legend-item-0"
              style="display: inline-block; margin-right: 10px;"
            >
              <svg
                class="recharts-surface"
                height="14"
                style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                version="1.1"
                viewBox="0 0 32 32"
                width="14"
              >
                <path
                  class="recharts-legend-icon"
                  d="M0,4h32v24h-32z"
                  fill="#8884d8"
                  stroke="none"
                />
              </svg>
              <span
                class="recharts-legend-item-text"
              >
                principal
              </span>
            </li>
            <li
              class="recharts-legend-item legend-item-1"
              style="display: inline-block; margin-right: 10px;"
            >
              <svg
                class="recharts-surface"
                height="14"
                style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                version="1.1"
                viewBox="0 0 32 32"
                width="14"
              >
                <path
                  class="recharts-legend-icon"
                  d="M0,4h32v24h-32z"
                  fill="#82ca9d"
                  stroke="none"
                />
              </svg>
              <span
                class="recharts-legend-item-text"
              >
                interest
              </span>
            </li>
          </ul>
        </div>
        <div
          class="recharts-tooltip-wrapper"
          style="transform: translate(undefinedpx, undefinedpx); pointer-events: none; visibility: hidden; position: absolute; top: 0px;"
        >
          <div
            class="recharts-default-tooltip"
            style="margin: 0px; padding: 10px; background-color: rgb(255, 255, 255); border: 1px solid #ccc; white-space: nowrap;"
          >
            <p
              class="recharts-tooltip-label"
              style="margin: 0px;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
